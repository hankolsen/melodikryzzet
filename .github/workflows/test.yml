name: melodikryzzet-tests

on:
  push:
    branches:
      - 'master'
  pull_request:

jobs:
  test:
    name: 'Node ${{ matrix.node }}'

    runs-on: ubuntu-latest

    strategy:
      matrix:
        node: [14, 16]

    env:
      SKIP_PREFLIGHT_CHECK: true

    steps:
      - uses: actions/checkout@v2

      - uses: actions/setup-node@v1
        with:
          node-version: ${{ matrix.node }}

      - name: Install dependencies
        uses: bahmutov/npm-install@v1
        with:
          install-command: yarn

      - name: Run tests
        run: yarn test:ci

  cypress-run:
    runs-on: ubuntu-latest

    env:
      SKIP_PREFLIGHT_CHECK: true
      NETLIFY_AUTH_TOKEN: ${{ secrets.NETLIFY_AUTH_TOKEN }}
      DB_NAME: ${{ secrets.DB_NAME }}
      CROSSWORD_COLLECTION_NAME: ${{ secrets.CROSSWORD_COLLECTION_NAME }}
      DB_USERNAME: ${{ secrets.DB_USERNAME }}
      DB_PASSWORD: ${{ secrets.DB_PASSWORD }}
      DB_HOST: ${{ secrets.DB_HOST }}

    steps:
      - uses: actions/checkout@v2

      - name: Cypress run
        uses: cypress-io/github-action@v2
        with:
          start: netlify dev
          wait-on: 'http://localhost:8888'
