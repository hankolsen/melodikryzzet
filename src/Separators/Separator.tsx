import React from 'react';
import { Direction, Position } from 'Crossword/Crossword.types';

import Line from './Line';

type Props = {
  direction: Direction;
  position: Position;
  separator: string;
  locations: number[];
  id: string;
};

const Separator = ({
  direction,
  position,
  separator,
  locations,
  id,
}: Props) => {
  switch (separator) {
    case ',':
      return (
        <>
          {locations.map((location) => (
            <Line
              key={`${id}-${location}`}
              position={position}
              location={location}
              direction={direction}
            />
          ))}
        </>
      );
    default:
      return null;
  }
};

export default Separator;
