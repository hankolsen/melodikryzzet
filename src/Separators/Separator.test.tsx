import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { Direction } from 'Crossword/Crossword.types';
import { render } from 'test/test-utils';
import Separator from './Separator';

describe('Separator test', () => {
  it('should not render anything for non-existing separator', () => {
    const { container } = render(
      <svg>
        <Separator
          direction={Direction.across}
          position={{ x: 3, y: 4 }}
          locations={[2]}
          id="2"
          separator="."
        />
      </svg>,
    );
    expect(container.querySelector('line')).not.toBeInTheDocument();
  });

  it('should render a line', () => {
    const { container } = render(
      <svg>
        <Separator
          direction={Direction.across}
          position={{ x: 2, y: 3 }}
          locations={[2]}
          id="3"
          separator=","
        />
      </svg>,
    );
    expect(container.querySelector('line')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
