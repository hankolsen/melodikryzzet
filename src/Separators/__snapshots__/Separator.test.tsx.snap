// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator test should render a line 1`] = `
<div>
  <div>
    <svg>
      <line
        stroke="black"
        stroke-width="4"
        x1="122.5"
        x2="133.5"
        y1="111.5"
        y2="111.5"
      />
    </svg>
  </div>
</div>
`;
